src/dataset/seeds.py

```diff
@@ -0,0 +1,696 @@
+        - color-type: seeds that involve an seen combination of color and type of object.
+        - agent-loc: seeds that involve an seen agent start position in the room or maze.
+        - object-task: seeds that involve an seen combination of object and task.
+        - rel-loc: seeds that involve an seen goal location relative to the agent.
+        - task-task: seeds (and envs/configs) that involve seen task combinations in sequence tasks.
+
+        Parameters
+        ----------
+            config (str): environment configuration name.
+
+        Returns
+        -------
+            list: list of seeds for a given environment configuration.
+        """
+
+        n_seeds = 100
+        seed_list = []
+        for seed in tqdm(range(0, 1000)):
+            env = gym.make(config)
+            env.reset(seed=seed)
+            if (
+                not self._check_size(env)
+                and not self._check_color_type(env)
+                and not self._check_object_task(env)
+                and not self._check_agent_loc(env)
+                and not self._check_rel_loc(env)
+                and not self._check_task_task(env)
+            ):
+                seed_list.append(seed)
+                if len(seed_list) == n_seeds:
+                    break
+
+        return seed_list
+
+    def _find_ood_seeds(self, config):
+        """
+        Find in-domain and ood seeds for the given environment and config.
+
+        Seeds are out-of-domain with respect any or all of the below:
+        - size: seeds that involve an unseen room or maze size.
+        - color-type: seeds that involve an unseen combination of color and type of object.
+        - object-task: seeds that involve an unseen combination of object and task.
+        - agent-loc: seeds that involve an unseen agent start position in the room or maze.
+        - rel-loc: seeds that involve an unseen goal location relative to the agent.
+        - task-task: seeds (and envs/configs) that involve unseen task combinations in sequence tasks.
+
+        Parameters
+        ----------
+            config (str): environment configuration name.
+            num_configs (int): number of configs that exist for a given environment.
+
+        Returns
+        -------
+            list: list of seeds for a given environment configuration.
+        """
+        seed_lists = {ood_type: [] for ood_type in self.ood_types}
+
+        n_seeds = 10
+        print(config)
+        for ood_type, seed_list in seed_lists.items():
+            if ood_type in ["object_task", "task_task"]:
+                max_seeds_to_check = 1000
+            else:
+                max_seeds_to_check = 100
+            for seed in tqdm(range(0, 10000)):
+                if seed >= max_seeds_to_check and len(seed_list) == 0:
+                    break
+                env = gym.make(config)
+                env.reset(seed=seed)
+                if ood_type == "size":
```

Comment:
Mostly a stylistic choice: I would rewrite this long list of "if" with a single "switch". In Python: https://stackoverflow.com/questions/21962763/using-a-dictionary-as-a-switch-statement-in-python